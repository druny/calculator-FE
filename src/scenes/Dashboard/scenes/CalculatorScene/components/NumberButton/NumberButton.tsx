import React, { Dispatch } from 'react';
import { Button } from '@material-ui/core';

type NumberButtonPropsType = {
  number: number;
  onClick: Dispatch<number>;
};

export function NumberButton({ number, onClick }: NumberButtonPropsType) {
  return (
    <Button variant="outlined" color="secondary" onClick={() => onClick(number)}>
      {number}
    </Button>
  );
}
