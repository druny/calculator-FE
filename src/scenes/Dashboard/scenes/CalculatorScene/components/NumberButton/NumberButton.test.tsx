import React from 'react';
import renderer from 'react-test-renderer';

import { NumberButton } from './NumberButton';

it('renders correctly when there are no items', () => {
  const tree = renderer.create(<NumberButton number={1} onClick={jest.fn()} />).toJSON();
  expect(tree).toMatchSnapshot();
});
