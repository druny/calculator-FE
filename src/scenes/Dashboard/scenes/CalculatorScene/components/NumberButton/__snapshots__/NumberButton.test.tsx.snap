// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    1
  </span>
</button>
`;
